src/main/java/org/tugraz/sysds/runtime/instructions/spark/BinarySPInstruction.java

```diff
@@ -160,7 +174,43 @@ protected void processMatrixMatrixBinaryInstruction(ExecutionContext ec)
 		sec.addLineageRDD(output.getName(), input2.getName());
 	}
 
-	protected void processMatrixBVectorBinaryInstruction(ExecutionContext ec, VectorType vtype) 
+	/**
+	 * Common binary tensor-tensor process instruction
+	 *
+	 * @param ec execution context
+	 */
+	protected void processTensorTensorBinaryInstruction(ExecutionContext ec) {
+		SparkExecutionContext sec = (SparkExecutionContext)ec;
+
+		//sanity check dimensions
+		checkTensorTensorBinaryCharacteristics(sec);
+		updateBinaryTensorOutputDataCharacteristics(sec);
+
+		// Get input RDDs
+		JavaPairRDD<TensorIndexes, TensorBlock> in1 = sec.getBinaryTensorBlockRDDHandleForVariable(input1.getName());
+		JavaPairRDD<TensorIndexes, TensorBlock> in2 = sec.getBinaryTensorBlockRDDHandleForVariable(input2.getName());
+		DataCharacteristics dcOut = sec.getDataCharacteristics(output.getName());
+
+		BinaryOperator bop = (BinaryOperator) _optr;
+
+		// TODO subtensor replication if required
```

Comment:
yes, please a note on splitting inputs with fewer dimensions into the appropriate blocking scheme.

Reply:
I actually just implemented the functionality itself, so you might want to take a look at that part if the implementation is fine.